sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/SecretsRegexTest.java

```diff
@@ -0,0 +1,195 @@
+import org.sonar.java.checks.regex.RegexComplexityCheck;
+import org.sonar.java.checks.regex.RegexLookaheadCheck;
+import org.sonar.java.checks.regex.RegexStackOverflowCheck;
+import org.sonar.java.checks.regex.ReluctantQuantifierCheck;
+import org.sonar.java.checks.regex.ReluctantQuantifierWithEmptyContinuationCheck;
+import org.sonar.java.checks.regex.SingleCharCharacterClassCheck;
+import org.sonar.java.checks.regex.SingleCharacterAlternationCheck;
+import org.sonar.java.checks.regex.SuperfluousCurlyBraceCheck;
+import org.sonar.java.checks.regex.UnicodeAwareCharClassesCheck;
+import org.sonar.java.checks.regex.UnicodeCaseCheck;
+import org.sonar.java.checks.regex.UnquantifiedNonCapturingGroupCheck;
+import org.sonar.java.checks.regex.UnusedGroupNamesCheck;
+import org.sonar.java.checks.regex.VerboseRegexCheck;
+import org.sonar.java.model.InternalSyntaxToken;
+import org.sonar.java.model.expression.IdentifierTreeImpl;
+import org.sonar.java.model.expression.LiteralTreeImpl;
+import org.sonar.java.model.expression.MemberSelectExpressionTreeImpl;
+import org.sonar.java.model.expression.MethodInvocationTreeImpl;
+import org.sonar.java.regex.JavaAnalyzerRegexSource;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.secrets.api.SpecificationLoader;
+import org.sonar.plugins.secrets.configuration.model.Rule;
+import org.sonar.plugins.secrets.configuration.model.matching.AuxiliaryPattern;
+import org.sonar.plugins.secrets.configuration.model.matching.BooleanCombination;
+import org.sonarsource.analyzer.commons.regex.RegexParser;
+import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
+
+class SecretsRegexTest {
+
+  private final static String CONFIGURATION_FILES_PATH = String.join(File.separator, "src", "main", "resources", "org", "sonar", "plugins", "secrets", "configuration");
+
+  private final TestRegexScannerContext context = new TestRegexScannerContext();
+
+
+  @Test
+  void shouldValidateAllRegexesInConfiguration() {
+    Path specificationsDirectory = Path.of(CONFIGURATION_FILES_PATH);
+    String[] extensionsToSearchFor = new String[]{"yaml"};
+    Collection<File> files = FileUtils.listFiles(new File(specificationsDirectory.toUri()), extensionsToSearchFor, false);
+
+    Set<String> listOfFileNames = files.stream().map(File::getName).collect(Collectors.toSet());
+
+    var specificationLoader = new SpecificationLoader(SpecificationLoader.DEFAULT_SPECIFICATION_LOCATION, listOfFileNames);
+
+    var ruleToRegexes = specificationLoader.getRulesMappedToKey().entrySet().stream()
+      .map(entry -> {
+
+        List<String> regexes = entry.getValue().stream()
+          .map(this::ruleToRegexes)
+          .flatMap(Collection::stream)
+          .collect(Collectors.toList());
+
+        return new Pair(entry.getKey(), regexes);
```

Comment:
Tbh. I'm not sure why we need the `Pair` class, I think we could do the same with a Map.